internal/gcloud/gcs/bucket.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcs
+
+import (
+	"io"
+
+	"golang.org/x/net/context"
+)
+
+// Bucket represents a GCS bucket, pre-bound with a bucket name and necessary
+// authorization information.
+//
+// Each method that may block accepts a context object that is used for
+// deadlines and cancellation. Users need not package authorization information
+// into the context object (using cloud.WithContext or similar).
+//
+// All methods are safe for concurrent access.
+type Bucket interface {
```

Comment:
the files under gcloud/gcs should go into internal/storage

Reply:
Ohh then I'll raise separate PR for moving the whole gcloud folder to internal/storage.

Reply:
yes. We already have one package for gcs interaction, then why are we creating seperate package again.

Reply:
Okay I'll raise separate PR for that.